Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /home/faffonso/Documentos/IC/GazeboSimulation/catkin_ws/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_5d363/fast && /usr/bin/make  -f CMakeFiles/cmTC_5d363.dir/build.make CMakeFiles/cmTC_5d363.dir/build
make[1]: Entrando no diretório '/home/faffonso/Documentos/IC/GazeboSimulation/catkin_ws/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_5d363.dir/src.c.o
/usr/bin/cc -DCMAKE_HAVE_LIBC_PTHREAD   -o CMakeFiles/cmTC_5d363.dir/src.c.o -c /home/faffonso/Documentos/IC/GazeboSimulation/catkin_ws/build/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_5d363
/home/faffonso/.local/lib/python3.8/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_5d363.dir/link.txt --verbose=1
/usr/bin/cc -rdynamic CMakeFiles/cmTC_5d363.dir/src.c.o -o cmTC_5d363 
/usr/bin/ld: CMakeFiles/cmTC_5d363.dir/src.c.o: na função "main":
src.c:(.text+0x46): referência não definida para "pthread_create"
/usr/bin/ld: src.c:(.text+0x52): referência não definida para "pthread_detach"
/usr/bin/ld: src.c:(.text+0x5e): referência não definida para "pthread_cancel"
/usr/bin/ld: src.c:(.text+0x6f): referência não definida para "pthread_join"
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_5d363.dir/build.make:99: cmTC_5d363] Erro 1
make[1]: Saindo do diretório '/home/faffonso/Documentos/IC/GazeboSimulation/catkin_ws/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:127: cmTC_5d363/fast] Erro 2


Source file was:
#include <pthread.h>

static void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/faffonso/Documentos/IC/GazeboSimulation/catkin_ws/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_e711a/fast && /usr/bin/make  -f CMakeFiles/cmTC_e711a.dir/build.make CMakeFiles/cmTC_e711a.dir/build
make[1]: Entrando no diretório '/home/faffonso/Documentos/IC/GazeboSimulation/catkin_ws/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_e711a.dir/CheckFunctionExists.c.o
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create -o CMakeFiles/cmTC_e711a.dir/CheckFunctionExists.c.o -c /home/faffonso/.local/lib/python3.8/site-packages/cmake/data/share/cmake-3.24/Modules/CheckFunctionExists.c
Linking C executable cmTC_e711a
/home/faffonso/.local/lib/python3.8/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_e711a.dir/link.txt --verbose=1
/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=pthread_create -rdynamic CMakeFiles/cmTC_e711a.dir/CheckFunctionExists.c.o -o cmTC_e711a  -lpthreads 
/usr/bin/ld: não foi possível localizar -lpthreads
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_e711a.dir/build.make:99: cmTC_e711a] Erro 1
make[1]: Saindo do diretório '/home/faffonso/Documentos/IC/GazeboSimulation/catkin_ws/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:127: cmTC_e711a/fast] Erro 2



